lasagne/layers/conv.py

```diff
@@ -781,3 +782,153 @@ def convolve(self, input, **kwargs):
+        The nonlinearity that is applied to the layer activations. If None
+        is provided, the layer will be linear.
+
+    flip_filters : bool (default: False)
+        Whether to flip the filters before sliding them over the input,
+        performing a convolution, or not to flip them and perform a
+        correlation (this is the default).
+        This implementation does not support flipped filters, the argument is
+        provided for compatibility to other convolutional layers only.
+
+    **kwargs
+        Any additional keyword arguments are passed to the `Layer` superclass.
+
+    Attributes
+    ----------
+    W : Theano shared variable or expression
+        Variable or expression representing the filter weights.
+
+    b : Theano shared variable or expression
+        Variable or expression representing the biases.
+
+    Notes
+    -----
+    The dilated convolution is implemented as the backward pass of a
+    convolution wrt. weights, passing the filters as the output gradient.
+    It can be thought of as dilating the filters (by adding ``dilation - 1``
+    zeros between adjacent filter elements) and cross-correlating them with the
+    input. See [1]_ for more background.
+
+    References
+    ----------
+    .. [1] Fisher Yu, Vladlen Koltun (2016),
+           Multi-Scale Context Aggregation by Dilated Convolutions. ICLR 2016.
+           http://arxiv.org/abs/1511.07122, https://github.com/fyu/dilation
+    """
+    def __init__(self, incoming, num_filters, filter_size, dilation=(1, 1),
+                 pad=0, untie_biases=False,
+                 W=init.GlorotUniform(), b=init.Constant(0.),
+                 nonlinearity=nonlinearities.rectify, flip_filters=False,
+                 **kwargs):
+        super(DilatedConv2DLayer, self).__init__(
+                incoming, num_filters, filter_size, dilation, pad,
+                untie_biases, W, b, nonlinearity, flip_filters, n=2, **kwargs)
+        # rename self.stride to self.dilation:
+        self.dilation = self.stride
+        del self.stride
+        # require valid convolution
+        if self.pad != (0, 0):
+            raise NotImplementedError(
+                    "DilatedConv2DLayer requires pad=0 / (0,0) / 'valid', but "
+                    "got %r. For a padded dilated convolution, add a PadLayer."
+                    % (pad,))
+        # require unflipped filters
+        if self.flip_filters:
+            raise NotImplementedError(
+                    "DilatedConv2DLayer requires flip_filters=False.")
+
+    def get_W_shape(self):
+        num_input_channels = self.input_shape[1]
+        # first two sizes are swapped compared to a forward convolution
+        return (num_input_channels, self.num_filters) + self.filter_size
+
+    def get_output_shape_for(self, input_shape):
+        # when called from the constructor, dilation is still called stride:
+        dilation = getattr(self, 'dilation', getattr(self, 'stride', None))
```

Comment:
Consider calling `super` in the constructor *after* renaming `stride`, so that you don't need this line.